rdmd/args.d

```diff
@@ -0,0 +1,29 @@
+/**
+Provides functionality for parsing command-line arguments and data
+structures for storing the results
+*/
+module rdmd.args;
+
+import rdmd.config : RDMDConfig;
+
+/**
+'Namespace' struct to encapsulate all global settings derived
+from command-line arguments to `rdmd`
+*/
+struct RDMDGlobalArgs
```

Comment:
Maybe just `Global`?

Reply:
Yes, there’s no reason for the `RDMD` prefix since it’s already in the `rdmd` module.

Reply:
Well, the `RDMD...` names are not chosen entirely arbitrarily.  There are a few good reasons for this formulation.

First, it really makes these symbols _highly_ visible.  It's not something you can just glance over and miss: the names are intrusive, and deliberately so, because (especially in the case of `RDMDGlobalArgs`) they are often being used in a way that is very problematic.  In-your-face visibility of a superficially "ugly" symbol name is a very good way to call attention to that.

Second, there's a slightly bigger-picture reason.  One of the goals of the refactoring effort is to more cleanly separate out all the stages of rdmd's decision making process.  So, for example, arguments parsing should be one clean process; preparing the parameters and settings for the build job should be a second process; and so on.

The `RDMD...` prefix is a nice way to highlight the data structures that are intended to be the result of one stage of rdmd's procedures, and the input to the next.